core/serialization/workspaces.js

```diff
@@ -28,31 +27,13 @@ const variables = goog.require('Blockly.serialization.variables');
  */
 const save = function(workspace) {
   const state = Object.create(null);
-
-  // TODO: Switch this to use plugin serialization system (once it is built).
-  const variableStates = [];
-  const vars = workspace.getAllVariables();
-  for (let i = 0; i < vars.length; i++) {
-    variableStates.push(variables.save(vars[i]));
-  }
-  if (variableStates.length) {
-    state['variables'] = variableStates;
-  }
-
-  const blockStates = [];
-  for (let block of workspace.getTopBlocks(false)) {
-    const blockState = blocks.save(block, {addCoordinates: true});
-    if (blockState) {
-      blockStates.push(blockState);
+  const serializerMap = registry.getAllItems(registry.Type.PLUGIN_SERIALIZER);
+  for (const key in serializerMap) {
+    const save = serializerMap[key].save(workspace);
+    if (save) {
+      state[key] = save;
```

Comment:
Does this mean for the blocks this would look like
```
'blocks': {
   languageVersion: 0, 
   blocks: [...]
}

Reply:
Yes it does.